core/src/main/java/org/elasticsearch/indices/analysis/HunspellService.java

```diff
@@ -70,8 +71,9 @@
  */
 public class HunspellService extends AbstractComponent {
 
-    public final static String HUNSPELL_LAZY_LOAD = "indices.analysis.hunspell.dictionary.lazy";
-    public final static String HUNSPELL_IGNORE_CASE = "indices.analysis.hunspell.dictionary.ignore_case";
+    public final static Setting<Boolean> HUNSPELL_LAZY_LOAD = Setting.boolSetting("indices.analysis.hunspell.dictionary.lazy", Boolean.FALSE, false, Setting.Scope.CLUSTER);
+    public final static Setting<Boolean> HUNSPELL_IGNORE_CASE = Setting.boolSetting("indices.analysis.hunspell.dictionary.ignore_case", Boolean.FALSE, false, Setting.Scope.CLUSTER);
+    public final static Setting<Settings> HUNSPELL_DICTIONARY_OPTIONS = Setting.groupSetting("indices.analysis.hunspell.dictionary.", false, Setting.Scope.CLUSTER);
     private final static String OLD_HUNSPELL_LOCATION = "indices.analysis.hunspell.dictionary.location";
```

Comment:
It will throw an exception if the setting is set but not registered in the ClusterSettings, right ?
We'll loose the explanation but is it really helpful:

''''
 String location = settings.get(OLD_HUNSPELL_LOCATION, null);
        if (location != null) {
            throw new IllegalArgumentException("please, put your hunspell dictionaries under config/hunspell !");
        }
"""